src/diagonal.works/b6/api/functions/collections.go

```diff
@@ -341,3 +342,155 @@ var _ api.Collection = &flattenCollection{}
 func flatten(_ *api.Context, c api.Collection) (api.Collection, error) {
 	return &flattenCollection{c: c}, nil
 }
+
+const maxBuckets = 5
+
+type bucket struct {
+	start, end interface{}
+}
+
+func bucketRange(b bucket) interface{} {
+	if b.start == b.end || b.end == nil {
+		return b.start
+	} else {
+		return fmt.Sprint(b.start) + "-" + fmt.Sprint(b.end)
+	}
+}
+
+type bucketFn func(interface{}, bucket) bool // Indicates whether given value can be bucketed.
+
+func bucketIndex(v interface{}, b []bucket, fn bucketFn) (int, error) {
+	for i, bucket := range b {
+		if fn(v, bucket) {
+			return i, nil
+		}
+	}
+
+	return -1, fmt.Errorf("can't bucket %v", v)
+}
+
+type kv struct {
+	key   interface{}
+	value int // count
+}
+
+func anyBuckets(kvs []kv) ([]bucket, error) {
+	sort.Slice(kvs, func(i, j int) bool {
+		return kvs[i].value > kvs[j].value
+	})
+
+	var buckets []bucket
+	for i, kv := range kvs {
+		if i == maxBuckets {
+			break
+		}
+
+		if i == maxBuckets-1 && len(kvs) > maxBuckets {
+			buckets = append(buckets, bucket{start: "other"})
+		} else {
+			buckets = append(buckets, bucket{start: kv.key})
+		}
+	}
+
+	return buckets, nil
+}
+
+func anyBucketFn(v interface{}, b bucket) bool {
+	return b.start == v || b.start == "other"
+}
+
+func uniformBuckets(kvs []kv) ([]bucket, error) {
+	sort.Slice(kvs, func(i, j int) bool {
+		iKey, _ := api.ToFloat64(kvs[i].key) // a: not dealing w errors here either, relying on ConvertibleToFloat64 check in histogram
```

Comment:
I think the standard library pattern for this, which we use in a few places, is to create a MustToFloat64() which doesn't return an error, but instead panic()s, and comment that line that it's checked earlier.